diff --git a/images/Makefile.common.in b/images/Makefile.common.in
--- a/images/Makefile.common.in
+++ b/images/Makefile.common.in
@@ -29,8 +29,8 @@ PLATFORMS?=linux/amd64,linux/arm64
 OUTPUT?=
 PROGRESS=auto
 EXTRA_BUILD_OPT?=
-build: ensure-buildx
-	docker buildx build $(if $(PLATFORMS),--platform=$(PLATFORMS),) $(OUTPUT) --progress=$(PROGRESS) -t ${IMAGE} --pull --build-arg GO_VERSION=$(GO_VERSION) $(EXTRA_BUILD_OPT) .
+build:
+	docker buildx build $(if $(PLATFORMS),--platform=$(PLATFORMS),) $(OUTPUT) --progress=$(PROGRESS) -t ${IMAGE} --build-arg GO_VERSION=$(GO_VERSION) $(EXTRA_BUILD_OPT) .
 
 # push the cross built image
 push: OUTPUT=--push
diff --git a/images/base/Dockerfile b/images/base/Dockerfile
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -20,7 +20,7 @@
 # start from debian slim, this image is reasonably small as a starting point
 # for a kubernetes node image, it doesn't contain much (anything?) we don't need
 # this stage will install basic files and packages
-ARG BASE_IMAGE=debian:bookworm-slim
+ARG BASE_IMAGE=debian:trixie-slim
 FROM $BASE_IMAGE AS base
 
 # copy in static files
@@ -109,10 +109,10 @@ COPY --chmod=0755 files/usr/local/bin/* /usr/local/bin/
 COPY --chmod=0755 scripts/third_party/gimme/gimme /usr/local/bin/
 COPY --chmod=0755 scripts/target-cc /usr/local/bin/
 # tools needed at build-time only
 # first ensure we can install packages for both architectures
-RUN dpkg --add-architecture arm64 && dpkg --add-architecture amd64 \
+RUN dpkg --add-architecture arm64 && dpkg --add-architecture amd64 && dpkg --add-architecture riscv64 \
     && clean-install bash ca-certificates curl git make pkg-config \
-    crossbuild-essential-amd64 crossbuild-essential-arm64 \
-    libseccomp-dev:amd64 libseccomp-dev:arm64
+    crossbuild-essential-amd64 crossbuild-essential-arm64 crossbuild-essential-riscv64 \
+    libseccomp-dev:amd64 libseccomp-dev:arm64 libseccomp-dev:riscv64
 # set by makefile to .go-version
 ARG GO_VERSION
 RUN eval "$(gimme "${GO_VERSION}")" \
diff --git a/images/base/files/etc/containerd/config.toml b/images/base/files/etc/containerd/config.toml
--- a/images/base/files/etc/containerd/config.toml
+++ b/images/base/files/etc/containerd/config.toml
@@ -34,7 +34,7 @@
 
 [plugins."io.containerd.grpc.v1.cri"]
   # use fixed sandbox image
-  sandbox_image = "registry.k8s.io/pause:3.10"
+  sandbox_image = "ghcr.io/go-riscv/pause:3.10.1-linux-riscv64"
   # allow hugepages controller to be missing
   # see https://github.com/containerd/cri/pull/1501
   tolerate_missing_hugepages_controller = true
diff --git a/images/base/scripts/target-cc b/images/base/scripts/target-cc
--- a/images/base/scripts/target-cc
+++ b/images/base/scripts/target-cc
@@ -18,8 +18,10 @@ set -o errexit -o nounset -o pipefail
 case $TARGETARCH in
   arm64)
     echo -n 'aarch64-linux-gnu-gcc' ;;
   amd64)
     echo -n 'x86_64-linux-gnu-gcc' ;;
+  riscv64)
+    echo -n 'riscv64-linux-gnu-gcc' ;;
   *)
     exit 1 ;;
 esac
diff --git a/images/kindnetd/Dockerfile b/images/kindnetd/Dockerfile
--- a/images/kindnetd/Dockerfile
+++ b/images/kindnetd/Dockerfile
@@ -33,7 +33,7 @@ RUN eval "$(gimme "${GO_VERSION}")" \
     && GOARCH=$TARGETARCH go-licenses save --save_path=/_LICENSES ./cmd/kindnetd
 
 # build real kindnetd image
-FROM registry.k8s.io/build-image/distroless-iptables:v0.7.4
+FROM ghcr.io/go-riscv/distroless-iptables-riscv64:v0.8.10
 COPY --from=0 --chown=root:root ./go/src/kindnetd /bin/kindnetd
 COPY --from=0 /_LICENSES/* /LICENSES/
 COPY --chmod=0644 files/LICENSES/* /LICENSES/*
diff --git a/images/local-path-helper/Dockerfile b/images/local-path-helper/Dockerfile
--- a/images/local-path-helper/Dockerfile
+++ b/images/local-path-helper/Dockerfile
@@ -15,7 +15,7 @@
 # This image is contains the binaries needed for the local-path-provisioner
 # helper pod. Currently that means: sh, rm, mkdir
 
-ARG BASE="registry.k8s.io/build-image/debian-base:bullseye-v1.4.3"
+ARG BASE="debian:trixie-slim"
 FROM ${BASE} AS build
 
 # NOTE: copyrights.tar.gz is a quirk of Kubernetes's debian-base image
@@ -42,6 +42,6 @@ RUN mkdir -p "${STAGE_DIR}" && \
     find "${STAGE_DIR}"
 
 # copy staged binary + deps + copyright into distroless
-FROM "gcr.io/distroless/static-debian11"
+FROM "gcr.io/distroless/static-debian13"
 ARG STAGE_DIR="/opt/stage"
 COPY --from=build "${STAGE_DIR}/" /
diff --git a/images/local-path-provisioner/Dockerfile b/images/local-path-provisioner/Dockerfile
--- a/images/local-path-provisioner/Dockerfile
+++ b/images/local-path-provisioner/Dockerfile
@@ -27,7 +27,7 @@ RUN eval "$(gimme "${GO_VERSION}")" \
     && mv bin/local-path-provisioner-$TARGETARCH /usr/local/bin/local-path-provisioner \
     && GOBIN=/usr/local/bin go install github.com/google/go-licenses@latest \
     && GOARCH=$TARGETARCH go-licenses save --save_path=/_LICENSES .
 
-FROM gcr.io/distroless/base-debian11
+FROM gcr.io/distroless/base-debian13
 COPY --from=0 /usr/local/bin/local-path-provisioner /usr/local/bin/local-path-provisioner
 COPY --from=0 /_LICENSES/* /LICENSES/
 COPY --chmod=0644 files/LICENSES/* /LICENSES/*
diff --git a/pkg/build/nodeimage/build.go b/pkg/build/nodeimage/build.go
--- a/pkg/build/nodeimage/build.go
+++ b/pkg/build/nodeimage/build.go
@@ -148,8 +148,9 @@ func supportedArch(arch string) bool {
 	default:
 		return false
 	// currently we nominally support building node images for these
 	case "amd64":
 	case "arm64":
+	case "riscv64":
 	}
 	return true
 }
diff --git a/pkg/build/nodeimage/const_cni.go b/pkg/build/nodeimage/const_cni.go
--- a/pkg/build/nodeimage/const_cni.go
+++ b/pkg/build/nodeimage/const_cni.go
@@ -17,7 +17,7 @@ package nodeimage
 /*
 The default CNI manifest and images are our own tiny kindnet
 */
 
-const kindnetdImage = "docker.io/kindest/kindnetd:v20251212-v0.29.0-alpha-105-g20ccfc88"
+const kindnetdImage = "ghcr.io/go-riscv/kindnetd:riscv64"
 
 var defaultCNIImages = []string{kindnetdImage}
diff --git a/pkg/build/nodeimage/const_storage.go b/pkg/build/nodeimage/const_storage.go
--- a/pkg/build/nodeimage/const_storage.go
+++ b/pkg/build/nodeimage/const_storage.go
@@ -26,8 +26,8 @@ NOTE: we have customized it in the following ways:
 - tolerate control plane scheduling taints
 */
 
-const storageProvisionerImage = "docker.io/kindest/local-path-provisioner:v20251212-v0.29.0-alpha-105-g20ccfc88"
-const storageHelperImage = "docker.io/kindest/local-path-helper:v20251211-v0.29.0-alpha-100-g82a92c5d"
+const storageProvisionerImage = "ghcr.io/go-riscv/local-path-provisioner:riscv64"
+const storageHelperImage = "ghcr.io/go-riscv/local-path-helper:riscv64"
 
 // image we need to preload
 var defaultStorageImages = []string{storageProvisionerImage, storageHelperImage}
