diff --git a/images/Makefile.common.in b/images/Makefile.common.in
--- a/images/Makefile.common.in
+++ b/images/Makefile.common.in
@@ -29,8 +29,8 @@ PLATFORMS?=linux/amd64,linux/arm64
 OUTPUT?=
 PROGRESS=auto
 EXTRA_BUILD_OPT?=
-build: ensure-buildx
-	docker buildx build $(if $(PLATFORMS),--platform=$(PLATFORMS),) $(OUTPUT) --progress=$(PROGRESS) -t ${IMAGE} --pull --build-arg GO_VERSION=$(GO_VERSION) $(EXTRA_BUILD_OPT) .
+build:
+	docker buildx build $(if $(PLATFORMS),--platform=$(PLATFORMS),) $(OUTPUT) --progress=$(PROGRESS) -t ${IMAGE} --build-arg GO_VERSION=$(GO_VERSION) $(EXTRA_BUILD_OPT) .

 # push the cross built image
 push: OUTPUT=--push
diff --git a/images/base/Dockerfile b/images/base/Dockerfile
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -107,10 +107,10 @@ COPY --chmod=0755 scripts/third_party/gimme/gimme /usr/local/bin/
 COPY --chmod=0755 scripts/target-cc /usr/local/bin/
 # tools needed at build-time only
 # first ensure we can install packages for both architectures
-RUN dpkg --add-architecture arm64 && dpkg --add-architecture amd64 \
+RUN dpkg --add-architecture arm64 && dpkg --add-architecture amd64 && dpkg --add-architecture riscv64 \
     && clean-install bash ca-certificates curl git make pkg-config \
-    build-essential crossbuild-essential-amd64 crossbuild-essential-arm64 \
-    libseccomp-dev:amd64 libseccomp-dev:arm64
+    build-essential crossbuild-essential-amd64 crossbuild-essential-arm64 crossbuild-essential-riscv64 \
+    libseccomp-dev:amd64 libseccomp-dev:arm64 libseccomp-dev:riscv64
 # set by makefile to .go-version
 ARG GO_VERSION
 RUN eval "$(gimme "${GO_VERSION}")" \
diff --git a/images/base/scripts/target-cc b/images/base/scripts/target-cc
--- a/images/base/scripts/target-cc
+++ b/images/base/scripts/target-cc
@@ -21,6 +21,8 @@ case $TARGETARCH in
     echo -n 'aarch64-linux-gnu-gcc' ;;
   amd64)
     echo -n 'x86_64-linux-gnu-gcc' ;;
+  riscv64)
+    echo -n 'gcc' ;;
   *)
     exit 1 ;;
 esac
diff --git a/images/kindnetd/Dockerfile b/images/kindnetd/Dockerfile
--- a/images/kindnetd/Dockerfile
+++ b/images/kindnetd/Dockerfile
@@ -31,7 +31,7 @@ RUN eval "$(gimme "${GO_VERSION}")" \
     && GOARCH=$TARGETARCH go-licenses save --save_path=/_LICENSES ./cmd/kindnetd

 # build real kindnetd image
-FROM registry.k8s.io/build-image/distroless-iptables:v0.8.7
+FROM distroless-iptables-riscv64:v0.8.10
 COPY --from=0 --chown=root:root ./go/src/kindnetd /bin/kindnetd
 COPY --from=0 /_LICENSES/* /LICENSES/
 COPY --chmod=0644 files/LICENSES/* /LICENSES/*
diff --git a/pkg/build/nodeimage/build.go b/pkg/build/nodeimage/build.go
--- a/pkg/build/nodeimage/build.go
+++ b/pkg/build/nodeimage/build.go
@@ -181,6 +181,7 @@ func supportedArch(arch string) bool {
 	case "amd64":
 	case "arm64":
+	case "riscv64":
 	}
 	return true
 }
