diff --git a/images/Makefile.common.in b/images/Makefile.common.in
--- a/images/Makefile.common.in
+++ b/images/Makefile.common.in
@@ -35,2 +35,2 @@
-build: ensure-buildx
-	docker buildx build $(if $(PLATFORMS),--platform=$(PLATFORMS),) $(OUTPUT) --progress=$(PROGRESS) -t ${IMAGE} --pull --build-arg GO_VERSION=$(GO_VERSION) $(EXTRA_BUILD_OPT) .
+build:
+	docker buildx build $(if $(PLATFORMS),--platform=$(PLATFORMS),) $(OUTPUT) --progress=$(PROGRESS) -t ${IMAGE} --build-arg GO_VERSION=$(GO_VERSION) $(EXTRA_BUILD_OPT) .
diff --git a/images/base/Dockerfile b/images/base/Dockerfile
--- a/images/base/Dockerfile
+++ b/images/base/Dockerfile
@@ -113 +113 @@
-RUN dpkg --add-architecture arm64 && dpkg --add-architecture amd64 \
+RUN dpkg --add-architecture arm64 && dpkg --add-architecture amd64 && dpkg --add-architecture riscv64 \
@@ -115,2 +115,2 @@
-    build-essential crossbuild-essential-amd64 crossbuild-essential-arm64 \
-    libseccomp-dev:amd64 libseccomp-dev:arm64
+    build-essential crossbuild-essential-amd64 crossbuild-essential-arm64 crossbuild-essential-riscv64 \
+    libseccomp-dev:amd64 libseccomp-dev:arm64 libseccomp-dev:riscv64
@@ -121 +121 @@
-    && GOBIN=/usr/local/bin go install github.com/google/go-licenses@latest
+    && GOBIN=/usr/local/bin go install github.com/google/go-licenses@v1.6.0
diff --git a/images/base/scripts/target-cc b/images/base/scripts/target-cc
--- a/images/base/scripts/target-cc
+++ b/images/base/scripts/target-cc
@@ -25,0 +26,2 @@
+  riscv64)
+    echo -n 'gcc' ;;
diff --git a/images/local-path-provisioner/Dockerfile b/images/local-path-provisioner/Dockerfile
--- a/images/local-path-provisioner/Dockerfile
+++ b/images/local-path-provisioner/Dockerfile
@@ -27,3 +27,2 @@
-    && GOARCH=$TARGETARCH scripts/build \
-    && mv bin/local-path-provisioner-$TARGETARCH /usr/local/bin/local-path-provisioner \
-    && GOBIN=/usr/local/bin go install github.com/google/go-licenses@latest \
+    && CGO_ENABLED=0 GOARCH=$TARGETARCH go build -ldflags "-X main.VERSION=${VERSION} -extldflags -static -s -w" -o /usr/local/bin/local-path-provisioner \
+    && GOBIN=/usr/local/bin go install github.com/google/go-licenses@v1.6.0 \
diff --git a/images/kindnetd/Dockerfile b/images/kindnetd/Dockerfile
--- a/images/kindnetd/Dockerfile
+++ b/images/kindnetd/Dockerfile
@@ -27 +27 @@
-    && GOBIN=/usr/local/bin go install github.com/google/go-licenses@latest
+    && GOBIN=/usr/local/bin go install github.com/google/go-licenses@v1.6.0
@@ -37 +37 @@
-FROM registry.k8s.io/build-image/distroless-iptables:v0.8.7
+FROM distroless-iptables-riscv64:v0.9.6
diff --git a/pkg/build/nodeimage/build.go b/pkg/build/nodeimage/build.go
--- a/pkg/build/nodeimage/build.go
+++ b/pkg/build/nodeimage/build.go
@@ -186,0 +187 @@
+	case "riscv64":
