diff --git a/images/build/cross/default/Dockerfile b/images/build/cross/default/Dockerfile
--- a/images/build/cross/default/Dockerfile
+++ b/images/build/cross/default/Dockerfile
@@ -32,2 +32,2 @@
-ARG KUBE_DYNAMIC_CROSSPLATFORMS="arm64 armhf i386 ppc64el s390x"
-ARG KUBE_CROSSPLATFORMS="linux/386 linux/arm linux/arm64 linux/ppc64le linux/s390x darwin/amd64 windows/amd64 windows/386"
+ARG KUBE_DYNAMIC_CROSSPLATFORMS="arm64 armhf i386 ppc64el riscv64 s390x"
+ARG KUBE_CROSSPLATFORMS="linux/386 linux/arm linux/arm64 linux/ppc64le linux/riscv64 linux/s390x darwin/amd64 windows/amd64 windows/386"
@@ -82 +82 @@
-    && apt-get install -y crossbuild-essential-ppc64el crossbuild-essential-s390x; \
+    && apt-get install -y crossbuild-essential-ppc64el crossbuild-essential-riscv64 crossbuild-essential-s390x; \
@@ -86 +86 @@
-  && if [ ${targetArch} = "ppc64le" ] || [ ${targetArch} = "s390x" ]; then \
+  && if [ ${targetArch} = "ppc64le" ] || [ ${targetArch} = "riscv64" ] || [ ${targetArch} = "s390x" ]; then \
@@ -93,17 +93,23 @@
-  && if [ ${targetArch} = "amd64" ]; then \
-  ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-x86_64.zip"; \
-elif [ ${targetArch} = "arm64" ]; then \
-  ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-aarch_64.zip"; \
-elif [ ${targetArch} = "ppc64le" ]; then \
-  ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-ppcle_64.zip"; \
-elif [ ${targetArch} = "s390x" ]; then \
-  ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-s390_64.zip"; \
-fi \
-  && mkdir /tmp/protoc && cd /tmp/protoc \
-  && wget "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/${ZIPNAME}" \
-  && unzip "${ZIPNAME}" \
-  && chmod -R +rX /tmp/protoc \
-  && cp -pr bin /usr/local \
-  && cp -pr include /usr/local \
-  && rm -rf /tmp/protoc \
-  && protoc --version
+  && if [ ${targetArch} = "riscv64" ]; then \
+    apt-get update \
+    && apt-get install -y protobuf-compiler \
+    && protoc --version; \
+  else \
+    if [ ${targetArch} = "amd64" ]; then \
+      ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-x86_64.zip"; \
+    elif [ ${targetArch} = "arm64" ]; then \
+      ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-aarch_64.zip"; \
+    elif [ ${targetArch} = "ppc64le" ]; then \
+      ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-ppcle_64.zip"; \
+    elif [ ${targetArch} = "s390x" ]; then \
+      ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-s390_64.zip"; \
+    fi \
+    && mkdir /tmp/protoc && cd /tmp/protoc \
+    && wget "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/${ZIPNAME}" \
+    && unzip "${ZIPNAME}" \
+    && chmod -R +rX /tmp/protoc \
+    && cp -pr bin /usr/local \
+    && cp -pr include /usr/local \
+    && rm -rf /tmp/protoc \
+    && protoc --version; \
+  fi
