diff --git a/images/build/cross/default/Dockerfile b/images/build/cross/default/Dockerfile
--- a/images/build/cross/default/Dockerfile
+++ b/images/build/cross/default/Dockerfile
@@ -34,6 +34,9 @@ ENV KUBE_DYNAMIC_CROSSPLATFORMS=${KUBE_DYNAMIC_CROSSPLATFORMS}
 ENV KUBE_CROSSPLATFORMS=${KUBE_CROSSPLATFORMS}
 
 ##------------------------------------------------------------
+
+# Copy precompiled artifacts that are awkward to fetch for riscv64.
+COPY precompiled /precompiled
 COPY --from=go-source /usr/local/go /usr/local/go
 ENV PATH=/usr/local/go/bin:$PATH
 ENV GOROOT=/usr/local/go
@@ -98,7 +101,10 @@ elif [ ${targetArch} = "s390x" ]; then \
   ZIPNAME="protoc-${PROTOBUF_VERSION}-linux-s390_64.zip"; \
 fi \
   && mkdir /tmp/protoc && cd /tmp/protoc \
-  && wget "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/${ZIPNAME}" \
+  && if [ -f "/precompiled/${ZIPNAME}" ]; then \
+       cp "/precompiled/${ZIPNAME}" .; \
+     else \
+       wget "https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOBUF_VERSION}/${ZIPNAME}"; fi \
   && unzip "${ZIPNAME}" \
   && chmod -R +rX /tmp/protoc \
   && cp -pr bin /usr/local \
