diff --git a/images/build/debian-base/Makefile b/images/build/debian-base/Makefile
--- a/images/build/debian-base/Makefile
+++ b/images/build/debian-base/Makefile
@@ -66,14 +66,6 @@ push-manifest:
 	docker manifest push --purge ${IMAGE}:${IMAGE_VERSION}
 
 build: clean
-	# Fix possible issues with the local umask
-	umask 0022
-
-	# Enable execution of multi-architecture containers
-	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) --reset -p yes
-	docker buildx version
-	BUILDER=$(shell docker buildx create --use)
-
 	docker buildx build \
 		--pull \
 		--load \
@@ -92,7 +84,6 @@ build: clean
 		-t $(IMAGE)-$(ARCH):$(TAG)-$(CONFIG) \
 		-t $(IMAGE)-$(ARCH):latest-$(CONFIG) \
 		$(CONFIG)
-	docker buildx rm $$BUILDER
 	rm $(CONFIG)/rootfs.tar
 
 push: build
