diff --git a/images/build/distroless-iptables/Makefile b/images/build/distroless-iptables/Makefile
--- a/images/build/distroless-iptables/Makefile
+++ b/images/build/distroless-iptables/Makefile
@@ -40,15 +40,7 @@ export DOCKER_CLI_EXPERIMENTAL := enabled
 SUDO=$(if $(filter 0,$(shell id -u)),,sudo)
 
 build:
-	# Fix possible issues with the local umask
-	umask 0022
-
-	# Enable execution of multi-architecture containers
-	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) --reset -p yes
-	docker buildx version
-	BUILDER=$(shell docker buildx create --use)
 	docker buildx build \
-		--pull \
 		--load \
 		--platform linux/$(ARCH) \
 		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION) \
@@ -58,7 +50,6 @@ build:
 		--build-arg=GORUNNERIMAGE=$(GORUNNERIMAGE) \
 		--build-arg=SKIP_WRAPPER_CHECK=$(SKIP_WRAPPER_CHECK) \
 		$(CONFIG)
-	docker buildx rm $$BUILDER
 
 push: build
 	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)
