diff --git a/hack/verify-build.sh b/hack/verify-build.sh
--- a/hack/verify-build.sh
+++ b/hack/verify-build.sh
@@ -25,0 +26 @@
+    linux/riscv64
diff --git a/images/build/cross/Makefile b/images/build/cross/Makefile
--- a/images/build/cross/Makefile
+++ b/images/build/cross/Makefile
@@ -56 +56 @@
-PLATFORMS ?= linux/amd64 linux/arm64 #linux/arm
+PLATFORMS ?= linux/amd64 linux/arm64 linux/riscv64 #linux/arm
@@ -58,2 +58,2 @@
-KUBE_DYNAMIC_CROSSPLATFORMS ?= "arm64 armhf i386 ppc64el s390x"
-KUBE_CROSSPLATFORMS ?= "linux/386 linux/arm linux/arm64 linux/ppc64le linux/s390x darwin/amd64 windows/amd64 windows/386"
+KUBE_DYNAMIC_CROSSPLATFORMS ?= "arm64 armhf i386 ppc64el riscv64 s390x"
+KUBE_CROSSPLATFORMS ?= "linux/386 linux/arm linux/arm64 linux/ppc64le linux/riscv64 linux/s390x darwin/amd64 windows/amd64 windows/386"
diff --git a/images/build/debian-base/Makefile b/images/build/debian-base/Makefile
--- a/images/build/debian-base/Makefile
+++ b/images/build/debian-base/Makefile
@@ -27 +27 @@
-ALL_ARCH = amd64 arm arm64 ppc64le s390x
+ALL_ARCH = amd64 arm arm64 ppc64le riscv64 s390x
@@ -49,0 +50,4 @@
+endif
+ifeq ($(ARCH),riscv64)
+	BASEIMAGE?=riscv64/debian:$(CONFIG)-slim
+	QEMUARCH=riscv64
diff --git a/images/build/distroless-iptables/Makefile b/images/build/distroless-iptables/Makefile
--- a/images/build/distroless-iptables/Makefile
+++ b/images/build/distroless-iptables/Makefile
@@ -27 +27 @@
-ALL_ARCH = amd64 arm arm64 ppc64le s390x
+ALL_ARCH = amd64 arm arm64 ppc64le riscv64 s390x
diff --git a/images/build/go-runner/Makefile b/images/build/go-runner/Makefile
--- a/images/build/go-runner/Makefile
+++ b/images/build/go-runner/Makefile
@@ -36 +36 @@
-PLATFORMS ?= linux/amd64 linux/arm64 linux/arm linux/ppc64le linux/s390x
+PLATFORMS ?= linux/amd64 linux/arm64 linux/arm linux/ppc64le linux/riscv64 linux/s390x
diff --git a/images/build/setcap/Makefile b/images/build/setcap/Makefile
--- a/images/build/setcap/Makefile
+++ b/images/build/setcap/Makefile
@@ -26 +26 @@
-ALL_ARCH = amd64 arm arm64 ppc64le s390x
+ALL_ARCH = amd64 arm arm64 ppc64le riscv64 s390x
