diff --git a/images/build/setcap/Makefile b/images/build/setcap/Makefile
--- a/images/build/setcap/Makefile
+++ b/images/build/setcap/Makefile
@@ -32,15 +32,7 @@ QEMUIMAGE ?= multiarch/qemu-user-static
 export DOCKER_CLI_EXPERIMENTAL := enabled
 
 build:
-	# Fix possible issues with the local umask
-	umask 0022
-
-	# Enable execution of multi-architecture containers
-	docker run --rm --privileged $(QEMUIMAGE):$(QEMUVERSION) --reset -p yes
-	docker buildx version
-	BUILDER=$(shell docker buildx create --use)
 	docker buildx build \
-		--pull \
 		--load \
 		--platform linux/$(ARCH) \
 		-t $(IMAGE)-$(ARCH):$(IMAGE_VERSION) \
@@ -48,7 +40,6 @@ build:
 		-t $(IMAGE)-$(ARCH):latest-$(CONFIG) \
 		--build-arg=BASEIMAGE=$(BASEIMAGE) \
 		.
-	docker buildx rm $$BUILDER
 
 push: build
 	docker push $(IMAGE)-$(ARCH):$(IMAGE_VERSION)
