diff --git a/Dockerfile-release.riscv64 b/Dockerfile-release.riscv64
new file mode 100644
--- /dev/null
+++ b/Dockerfile-release.riscv64
@@ -0,0 +1,11 @@
+FROM scratch
+
+ADD etcd /usr/local/bin/
+ADD etcdctl /usr/local/bin/
+ADD etcdutl /usr/local/bin/
+
+WORKDIR /var/lib/etcd/
+
+EXPOSE 2379 2380
+
+CMD ["/usr/local/bin/etcd"]
diff --git a/scripts/build-docker.sh b/scripts/build-docker.sh
--- a/scripts/build-docker.sh
+++ b/scripts/build-docker.sh
@@ -29,5 +29,8 @@
 ARCH=$(go env GOARCH)
 VERSION="${VERSION}-${ARCH}"
 DOCKERFILE="Dockerfile"
+if [ "${ARCH}" = "riscv64" ]; then
+  DOCKERFILE="Dockerfile-release.riscv64"
+fi
 
 if [ -z "${BINARYDIR:-}" ]; then
diff --git a/server/etcdmain/etcd.go b/server/etcdmain/etcd.go
--- a/server/etcdmain/etcd.go
+++ b/server/etcdmain/etcd.go
@@ -237,6 +237,6 @@ func checkSupportArch() {
 	// To add a new platform, check https://github.com/etcd-io/website/blob/main/content/en/docs/${VERSION}/op-guide/supported-platform.md.
 	// The ${VERSION} is the etcd version, e.g. v3.5, v3.6 etc.
 	switch runtime.GOARCH {
-	case "amd64", "arm64", "ppc64le", "s390x":
+	case "amd64", "arm64", "ppc64le", "riscv64", "s390x":
 		return
 	}
