diff --git a/hack/lib/golang.sh b/hack/lib/golang.sh
--- a/hack/lib/golang.sh
+++ b/hack/lib/golang.sh
@@ -27,2 +27,3 @@
   linux/ppc64le
+  linux/riscv64
 )
@@ -35,2 +36,3 @@
   linux/ppc64le
+  linux/riscv64
   windows/amd64
@@ -47,2 +49,3 @@
   linux/ppc64le
+  linux/riscv64
   darwin/amd64
@@ -61,2 +64,3 @@
   linux/ppc64le
+  linux/riscv64
   darwin/amd64
@@ -214,4 +218,4 @@
     host_arch=$(kube::util::host_arch)
-    if [[ "${host_arch}" != "amd64" && "${host_arch}" != "arm64" && "${host_arch}" != "ppc64le" && "${host_arch}" != "s390x" ]]; then
-      # on any platform other than amd64, arm64, ppc64le and s390x, we just default to amd64
+    if [[ "${host_arch}" != "amd64" && "${host_arch}" != "arm64" && "${host_arch}" != "ppc64le" && "${host_arch}" != "riscv64" && "${host_arch}" != "s390x" ]]; then
+      # on any platform other than amd64, arm64, ppc64le, riscv64 and s390x, we just default to amd64
       host_arch="amd64"
@@ -509,2 +513,6 @@
         ;;
+      "linux/riscv64")
+        export CGO_ENABLED=1
+        export CC=${KUBE_LINUX_RISCV64_CC:-riscv64-linux-gnu-gcc}
+        ;;
     esac
diff --git a/hack/lib/util.sh b/hack/lib/util.sh
--- a/hack/lib/util.sh
+++ b/hack/lib/util.sh
@@ -187,4 +187,7 @@
       ;;
+    riscv64*)
+      host_arch=riscv64
+      ;;
     *)
-      kube::log::error "Unsupported host arch. Must be x86_64, 386, arm, arm64, s390x or ppc64le."
+      kube::log::error "Unsupported host arch. Must be x86_64, 386, arm, arm64, s390x, ppc64le or riscv64."
       exit 1
diff --git a/hack/local-up-cluster.sh b/hack/local-up-cluster.sh
--- a/hack/local-up-cluster.sh
+++ b/hack/local-up-cluster.sh
@@ -411,4 +411,7 @@
         ;;
+      riscv64*)
+        host_arch=riscv64
+        ;;
       *)
-        echo "Unsupported host arch. Must be x86_64, 386, arm, arm64, s390x or ppc64le." >&2
+        echo "Unsupported host arch. Must be x86_64, 386, arm, arm64, s390x, ppc64le or riscv64." >&2
         exit 1
